pkg/collector/corechecks/containers/docker_rate_capping.go

```diff
@@ -0,0 +1,69 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+// +build docker
+
+package containers
+
+import (
+	log "github.com/cihub/seelog"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+)
+
+// cappedSender wraps around the standard Sender and overrides
+// the Rate method to implement rate capping
+type cappedSender struct {
+	aggregator.Sender
+	previousRateValues map[string]float64
+	rateCaps           map[string]float64
+}
+
+// Rate checks the rate value against the `capped_metrics` configuration
+// to filter out buggy spikes coming for cgroup cpu accounting
+func (s *cappedSender) Rate(metric string, value float64, hostname string, tags []string) {
+	capValue, found := s.rateCaps[metric]
+	if !found { // Metric not capped
+		s.Sender.Rate(metric, value, hostname, tags)
+		return
+	}
+	previousValue, found := s.previousRateValues[metric]
+	if !found { // First submit of the rate
+		s.previousRateValues[metric] = value
+		s.Sender.Rate(metric, value, hostname, tags)
+		return
+	}
+	delta := value - previousValue
```

Comment:
should we compute the rate here instead of the delta? I think when users set the cap they expect it to apply to the rate as they see it in the app.